### DISCUSSION TITLE
Cannot pass a dict type variable arg to Celery task when using Elasticsearch as result backend

### DISCUSSION THREAD

**Discussion by k0286** [celery/celery#7941] [Q&A] [answered]
Hello Celery community,
I switch my celery result backend from Redis to Elasticseach. 
And then I notice that some tasks will make elasticsearch  to raise exceptions that don't happen when using redis as result backend.
After some survey, I find out that is caused by tasks which accept args are dict type variables.

System Info
- Celery 5.2.7
- Elasticsearch 7.17.6
- Django 3.2

Celery Configuration
- result_extended `True`
- result_serializer `json`
- task_serializer `json`
- elasticsearch_save_meta_as_text `False`

Reproduce
1.  design the `tasks.py`
```python
from celery import shared_task

@shared_task()
def foo(a):
    return a
```
2. call the task and give a dict type variable as the argument
```python
a = {'foo': 'bar'}
foo.delay(a)  # or foo.apply_async((a,))
```
3. check the worker's log, it will get the exception as following
```shell
elasticsearch.exceptions.RequestError: RequestError(400, 'mapper_parsing_exception', "failed to parse field [result.args] of type [long] in document with id 'celery-task-meta-0ac700f2-a629-4af6-bf8f-1d5ed5351595'. Preview of field's value: '{foo=bar}'")
```

**Comment by auvipy** [comment]
what about checking elasticsearch_save_meta_as_text False to True?

**Reply by k0286** [reply]
Because I need to use the task's meta information in my apps, I have to set `elasticsearch_save_meta_as_text` to `False`  which make the task's meta information searchable.

**Reply by auvipy** [reply]
can you please open a PR with failing test case?

**Reply by k0286** [reply]
I'm not sure how to do it correctly. Is there any template or example that I can follow?

**Reply by auvipy** [reply]
https://github.com/celery/celery/blob/master/t/unit/backends/test_elasticsearch.py

**Comment by k0286** [comment, marked as answer]
After a couple days of survey, I realize that is caused by the restriction of [array type](https://www.elastic.co/guide/en/elasticsearch/reference/current/array.html) in the elasticsearch.

The elements of array in ES should be the same data type. 

So if my function's args are a string and a dict, the exception is raised.
